client/rpcserver/rpcserver.go

```diff
@@ -273,12 +274,7 @@ func (s *RPCServer) Run(ctx context.Context) {
 	// ctx passed to newMarketSyncer when making new market syncers.
 	s.ctx = ctx
 
-	// Create listener.
-	listener, err := tls.Listen("tcp", s.addr, s.tlsConfig)
-	if err != nil {
-		log.Errorf("can't listen on %s. rpc server quitting: %v", s.addr, err)
-		return
-	}
+	listener := createListener("tcp", s)
```

Comment:
Does moving this to a function add anything? If the code isn't being reused, I'm not certain it does.

Reply:
It allows the different createListener scenarios, fail/not fail, to be unit tested. It is not reused otherwise.